gwpy/table/io/pycbc.py

```diff
@@ -234,11 +234,21 @@ def empty_hdf5_file(h5f, ifo=None):
     # the decorator opens the HDF5 file for us, so h5f is guaranteed to
     # be an h5py.Group object
     h5f = h5f.file
+    # if the root group is empty, the file is empty
     if list(h5f) == []:
         return True
-    if ifo is not None and (ifo not in h5f or list(h5f[ifo]) == ['psd']):
-        return True
-    return False
+
+    # for each group (or the IFO group given by the user),
+    # check whether there is any useful content
+    if ifo is None:
```

Comment:
@duncanmmacleod, it may also appease codeclimate to replace this with a one-liner:

```python
    groups = [ifo] if ifo else h5f.keys()
```